src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.cs

```diff
@@ -176,6 +176,33 @@ public class MyClass { }
             }
         }
 
+        [Fact]
+        public async Task SucceedWhenGivenConflictingTypeNames()
+        {
+            // Regression test for https://github.com/dotnet/runtime/issues/93498
+
+            string source = """
+                using Microsoft.Extensions.Configuration;
+
+                var c = new ConfigurationBuilder().Build();
+                c.Get<Foo.Bar.BType>();
+
+                namespace Microsoft.Foo
+                {
+                    internal class AType {}
+                }
+
+                namespace Foo.Bar
+                {
+                    internal class BType {}
+                }
+                """;
+
+            ConfigBindingGenRunResult result = await RunGeneratorAndUpdateCompilation(source);
+            Assert.NotNull(result.GeneratedSource);
```

Comment:
>Assert.NotNull(result.GeneratedSource); [](http://example.com/codeflow?start=12&length=39)

would make sense to check the global names of both types in the generated source? I mean searching the text with these names? I am not feeling strongly about. just wanted to say it :-)

Reply:
I would consider this an implementation detail, from a testing perspective what matters is that the particular example compiles. I figured that the baselines should provide sufficient evidence as to what code does get generated.